cli/templatecheckout.go

```diff
@@ -0,0 +1,85 @@
+package cli
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/cli/cliui"
+)
+
+func TarBytesToTree(destination string, raw []byte) error {
```

Comment:
This function already exists in the codebase and can be found in `provisionersdk.Untar`. It's located there because eventual creators of provisions will likely have to interface with tars.

`cli.TarBytesToTree` would be confusing as an exposed function. It wouldn't be entirely clear why this function is located in the CLI package.

Reply:
Doh okay I'll look at that function.